src/ui/pages/AssetInfo/AssetInfo.tsx

```diff
@@ -0,0 +1,253 @@
+    handleScroll();
+    window.addEventListener('scroll', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
+  return showTokenInfoInHeader ? null : (
+    <AssetHeader asset={asset} className={styles.assetHeaderContent} />
+  );
+}
+
+function ReportAssetLink({ asset }: { asset: Asset }) {
+  return (
+    <UnstyledAnchor
+      target="_blank"
+      href={`https://zerion-io.typeform.com/to/IVsRHfBy?typeform-medium=embed-snippet#symbol=${asset.symbol}&asset_id=${asset.id}`}
+      rel="noopener noreferrer"
+      className="parent-hover"
+      style={{
+        ['--parent-content-color' as string]: 'var(--neutral-400)',
+        ['--parent-hovered-content-color' as string]: 'var(--neutral-700)',
+      }}
+    >
+      <HStack
+        gap={8}
+        alignItems="center"
+        className="content-hover"
+        justifyContent="center"
+      >
+        <FlagIcon style={{ width: 20, height: 20 }} />
+        <UIText kind="small/accent">Report Asset</UIText>
+      </HStack>
+    </UnstyledAnchor>
+  );
+}
+
+export function AssetInfo() {
+  const { asset_code } = useParams();
+  const navigationType = useNavigationType();
+  useEffect(() => {
+    if (navigationType === NavigationType.Push) {
+      window.scrollTo(0, 0);
+    }
+  }, [navigationType]);
+  useBackgroundKind({ kind: 'white' });
+  invariant(asset_code, 'Asset Code is required');
+
+  const { currency } = useCurrency();
+  const { data: assetFullInfoData } = useAssetFullInfo(
+    { currency, fungibleId: asset_code },
+    { source: useHttpClientSource() }
+  );
+  const assetFullInfo = assetFullInfoData?.data;
+  const { ready, singleAddress, singleAddressNormalized } = useAddressParams();
+  const { data: portfolioData } = useWalletPortfolio(
+    {
+      addresses: [singleAddressNormalized],
+      currency,
+      nftPriceType: 'not_included',
+    },
+    { source: useHttpClientSource() },
+    { enabled: ready }
+  );
+  const { data: walletData } = useWalletAssetDetails(
+    {
+      assetId: asset_code,
+      currency,
+      groupBy: ['by-app'],
+      addresses: [singleAddressNormalized],
+    },
+    { source: useHttpClientSource() },
+    { enabled: ready }
+  );
+
+  const { data: wallet } = useQuery({
+    queryKey: ['wallet/uiGetWalletByAddress', singleAddress, null],
+    queryFn: () =>
+      walletPort.request('uiGetWalletByAddress', {
+        address: singleAddress,
+        groupId: null,
+      }),
+    suspense: false,
+    enabled: ready,
+  });
+
+  const chainWithTheBiggestBalance = useMemo(() => {
+    const chainData = walletData?.data?.chainsDistribution;
+    if (!chainData) {
+      return NetworkId.Zero;
+    }
+    return chainData.reduce(
+      (acc, chain) =>
+        chain.value > acc.value
+          ? { value: chain.value, chain: chain.chain.id }
+          : acc,
+      { value: chainData[0].value ?? 0, chain: chainData[0].chain.id }
+    ).chain;
+  }, [walletData]);
```

Comment:
```suggestion
  const chainWithTheBiggestBalance = walletData?.data?.chainsDistribution.at(0)?.chain.id || NetworkId.Zero;
```

It looks as if backend returns this list already sorted.